README.md

```diff
@@ -1,486 +1,679 @@
+}
+
+C.prototype.method = logged(C.prototype.method, { kind: "method", name: "method", isStatic: false });
+x_setter = logged(x_setter, {kind: "setter", isStatic: false});
+```
+
+Note that getters and setters are decorated separately. Accessors are not "coalesced" as in earlier decorators proposals  (unless they are generated for a field; see below).
+
+## `@defineElement`
 
 [HTML Custom Elements](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements) lets you define your own HTML element. Elements are registered using `customElements.define`. Using decorators, the registration can happen up-front:
 
 ```mjs
-import { @defineElement } from "./defineElement.mjs";
+import { defineElement } from "./defineElement.mjs";
 
 @defineElement('my-class')
 class MyClass extends HTMLElement { }
 ```
 
-The `@defineElement` decorator is based on the `@register` decorator. This decorator is called when the class is finished being defined, and it calls the provided callback with the class that resulted.
+Classes can be decorated just like methods and accessors. The class shows up in the `value` option.
 
 ```mjs
 // defineElement.mjs
-export decorator @defineElement(name, options) {
-  @register(klass => customElements.define(name, klass, options))
+export function defineElement(name, options) {
+  return klass => { customElements.define(name, klass, options); return klass; }
 }
 ```
 
-This example uses a `decorator` declaration with a parameter list, which lets a decorator definition take arguments that can be used to supply arguments to other decorators in its definition.
+The decorator takes arguments at its usage site, so it is implemented as a function that returns another function. You can think of it as a "decorator factory": after you apply the arguments, it gives you another decorator.
 
-### `@metadata`
+This decorator usage could be desugared as follows:
 
-The `@metdata(key, value)` decorator is similar to [`@Reflect.metadata`](https://github.com/rbuckton/reflect-metadata): It allows the easy retrieval of information which was stored by annotating the class. The following examples are written in terms of the Reflect.metadata proposal, but you could imagine storing the metadata in any other form.
+```js
+class MyClass extends HTMLElement { }
+MyClass = defineElement(MyClass, {kind: "class"});
+```
 
-```mjs
-import { @metadata } from "./metadata.mjs";
+### Decorators adding metadata
 
-// partially apply the decorator locally for terseness
-decorator @localMeta { @metadata("key", "value") }
+Decorators can add metadata about class elements by adding a `metadata` property of the context object that is passed in to them. All of the metadata objects are `Object.assign`'ed together and placed in a property reachable from `[Symbol.metadata]` on the class. For example:
 
-@localMeta class C {
-  @localMeta method() { }
+```mjs
+@annotate({x: "y"}) @annotate({v: "w"}) class C {
```

Comment:
Isn't it `@{x: "y"}`?

Reply:
README has just the core proposal, which omits annotations.